app/models/concerns/profile_searching.rb

```diff
@@ -148,6 +156,25 @@ def filter_by_policy(_filter, _operator, policy_or_profile_id)
         conditions: "profiles.id IN (#{any_child_profile.to_sql})"
       }
     end
+
+    def filter_by_policy_and_profile_names(_filter, operator, value)
+      ss_qb = ScopedSearch::QueryBuilder.new(scoped_search_definition, '', scoped_search_definition.profile)
+      values = ss_qb.preprocess_parameters(
+        scoped_search_definition.field_by_name('name'), OPERATORS[operator], value
+      ) do |x|
+        # Passing empty block because nothing needs to happen on yield in preprocess_parameters
+        # The correct yield is invoked in to_ext_method_sql so all notification values are updated correctly
+        # See https://github.com/wvanbergen/scoped_search/blob/master/lib/scoped_search/query_builder.rb
+      end
```

Comment:
This looks a little weird but I couldn't think of a better way to do this. A block needs to be passed to `preprocess_parameters` because of the yield. [This](https://github.com/wvanbergen/scoped_search/blob/10d118690b6ba5a4941ac17e3d9157ddc817592f/lib/scoped_search/query_builder.rb#L56) is the block that's normally passed so it's just updating the `notification` values. 
When ext_methods are used, this is being handled [here in the to_ext_method_sql method](https://github.com/wvanbergen/scoped_search/blob/10d118690b6ba5a4941ac17e3d9157ddc817592f/lib/scoped_search/query_builder.rb#L422). So nothing needs to happen in the block but because of the yield, passing a block isn't optional.

Reply:
If you make an empty block, make it a one-liner and just add the comment explaining it before.